modules/core/src/main/java/org/apache/ignite/internal/processors/cache/verify/IdleVerifyUtility.java

```diff
@@ -109,6 +112,70 @@ public static boolean isCheckpointNow(@Nullable IgniteCacheDatabaseSharedManager
         return progress != null && progress.inProgress();
     }
 
+    /**
+     * Gather updateCounters info.
+     *
+     * @param ign Ignite instance.
+     * @param grpIds Group Id`s.
+     * @return Counters info.
+     */
+    public static T2<Set<Integer>, Map<Integer, Set<Map.Entry<Integer, Long>>>> updCountersSnapshot(
+        IgniteEx ign,
+        Set<Integer> grpIds
+    ) {
+        Map<Integer, Set<Map.Entry<Integer, Long>>> partsWithCountersPerGrp = new HashMap<>();
+
+        for (Integer grpId : grpIds) {
+            CacheGroupContext grpCtx = ign.context().cache().cacheGroup(grpId);
+
+            GridDhtPartitionTopology top = grpCtx.topology();
+
+            partsWithCountersPerGrp.putIfAbsent(grpId, new HashSet<>());
+
+            Set<Map.Entry<Integer, Long>> partsWithCounters = partsWithCountersPerGrp.get(grpId);
```

Comment:
partsWithCountersPerGrp.computeIfAbsent(grpId, () -> new HashSet<>()); may be better here. It allows to avoid getting element that was put in line above.